src/kernel.ts

```diff
@@ -342,10 +348,12 @@ class Kernel {
       }
       validateKernelId(success.data);
       this.connect();
-      return success.data;
     }, (error: utils.IAjaxError) => {
       this._onError(error);
+    }).catch((data?: any) => {
+      this._promiseDelegate.reject(data);
```

Comment:
What if the current promise is not settled, but is pending some other operation? The old one gets blow away, but the pending request operates on the new delegate.